File: csharp/series/Series.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Series
{
  private string _series;

  public Series(string input)
  {
    _series = input;
  }

  public int[][] Slices(int sliceLength)
  {
    if (sliceLength > _series.Length)
    {
      throw new ArgumentException("The length of a slice cannot exceed the length of the series.");
    }

    var result = new List<int[]>();

    for (int i = 0; i + sliceLength - 1 < _series.Length; i++)
    {
      result.Add(SplitSeries(_series.Substring(i, sliceLength)));
    }
    return result.ToArray();
  }

  private int[] SplitSeries(string input)
  {
    return input.Select(x => int.Parse(x.ToString())).ToArray();
  }
}

File: javascript/series/series.js
export class Series {
  constructor(series) {
    this._series = series;
  }

  slices(sliceLength) {
    if (this._series.length === 0) throw new Error('series cannot be empty');
    else if (sliceLength > this._series.length) throw new Error('slice length cannot be greater than series length');
    else if (sliceLength === 0) throw new Error('slice length cannot be zero');
    else if (sliceLength < 0) throw new Error('slice length cannot be negative');
    else {
      let result = [];
      for (let i = 0; i + sliceLength <= this._series.length; i++) {
        result.push(this._series.substring(i, i + sliceLength).split('').map(s => Number(s)));
      }
      return result;
    }
  }
}
